### PR TITLE
Widget container can now drag and drop widgets

### PR DESCRIPTION
**Pull request by samuelsan**

### CODE REVIEW

File: src/containers/old-dashboard/WidgetContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  generateWidget() {
+    const WidgetComponent = this.props.widget;
+    return <WidgetComponent/>
+  }
+
+  // for sectioning logic purposes perhaps
+  // @ts-ignore
+  refCallback = element => {
+    if (element) {
+      console.log("CALLBACK IS HAPPENING");
```
Comment (BhavikaSharma) [inline src/containers/old-dashboard/WidgetContainer.tsx]: Take these out.

File: src/containers/old-dashboard/WidgetContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  widgetID: any;
+  index: number;
+  widget: any;
+  key: number;
+}
+
+class WidgetContainer extends React.Component<WidgetContainerProps> {
+  // @ts-ignore
+  constructor(WidgetContainerProps) {
+    super(WidgetContainerProps);
```
Comment (BhavikaSharma) [inline src/containers/old-dashboard/WidgetContainer.tsx]: Idk React that well but is this line necessary? 

cc: @stanhuan

File: src/containers/old-dashboard/index.tsx [hunk 3]
```diff
@@ -34,10 +74,14 @@ class OldDashboard extends React.Component {
     }
 
     const startSection : any = (this as any).state.sections[source.droppableId];
-    // @ts-ignore
-    const finishSection = this.state.sections[destination.droppableId];
-
-    if (startSection === finishSection){
+    const finishSection : any = (this as any).state.sections[destination.droppableId];
+
+    console.log("start section");
```
Comment (BhavikaSharma) [inline src/containers/old-dashboard/index.tsx]: Remove these.

File: src/containers/old-dashboard/index.tsx [hunk 4]
```diff
@@ -34,10 +74,14 @@ class OldDashboard extends React.Component {
     }
 
     const startSection : any = (this as any).state.sections[source.droppableId];
-    // @ts-ignore
-    const finishSection = this.state.sections[destination.droppableId];
-
-    if (startSection === finishSection){
+    const finishSection : any = (this as any).state.sections[destination.droppableId];
```
Comment (BhavikaSharma) [inline src/containers/old-dashboard/index.tsx]: Can u add a comment as to what this is doing so people know in the future.

File: src/containers/old-dashboard/index.tsx [hunk 5]
```diff
@@ -59,15 +104,15 @@ class OldDashboard extends React.Component {
...
     }
 
-    const startWidgetIds = Array.from(startSection.widgetIds);
-    startWidgetIds.splice(source.index, 1);
-    const newStartWidgetIds = {
-      ...startSection,
-      widgetIds: startWidgetIds,
-    };
-
+    // const startWidgetIds = Array.from(startSection.widgetIds);
```
Comment (BhavikaSharma) [inline src/containers/old-dashboard/index.tsx]: If you need this for the future then it's fine but if it's just old code, please remove.

File: src/containers/old-dashboard/index.tsx [hunk 6]
```diff
@@ -76,13 +119,13 @@ class OldDashboard extends React.Component {
     const newState = {
         ...this.state,
         sections: {
+          // @ts-ignore
           ...this.state.sections,
-          [newStartWidgetIds.id]: newStartWidgetIds,
+          // [newStartWidgetIds.id]: newStartWidgetIds,
```
Comment (jj8huang) [inline src/containers/old-dashboard/index.tsx]: Should we be keeping this comment in here?

**Pull request merged**